src/cryptonote_core/blockchain.cpp

```diff
@@ -1774,11 +1848,10 @@ bool Blockchain::create_block_template_internal(
         diffic = get_difficulty_for_alternative_chain(alt_chain, bei.height, !info.is_miner);
     } else {
         // Creates the block template for next block on main chain
-        height = m_db->height();
+        std::tie(height, b.prev_id) = get_tail_id();
         auto [maj, min] = get_ideal_block_version(m_nettype, height);
         b.major_version = maj;
         b.minor_version = min;
-        b.prev_id = get_tail_id();
         median_weight = m_current_block_cumul_weight_limit / 2;
         diffic = get_difficulty_for_next_block(!info.is_miner);
         already_generated_coins = m_db->get_block_already_generated_coins(height - 1);
```

Comment:
This has to be

```cpp
already_generated_coins = m_db->get_block_already_generated_coins(height);
```

To get the cumulative generated coins (since we use `get_tail_id` which gives the height of the last block committed to the chain)

Reply:
On second thought instead of this suggestion, we have to amend this line here

```cpp
        std::tie(height, b.prev_id) = get_tail_id();
```

To convert the height into blockchain height

```cpp
        std::tie(height, b.prev_id) = get_tail_id();
        height += 1;
```

Because the height value returned from create block template and all other code that uses it, expects it to be the height of the blockchain (e.g. the height of the next block to be produced).

Reply:
Good catch; fixed.